### PR TITLE
feat(CX-40350): Add features

### PR DESCRIPTION
**Pull request by eltonplima**
### Summary
This PR introduces several new features and improvements to enhance the flexibility and usability of the `Plug.LoggerJSON` system for various logging scenarios. The changes include separate request/response logging control, customizable duration units, and better exception handling support.

#### 1. **Separate Request and Response Logging Control**
- Added support for independent control of request and response logging through two configurable functions
- Allows fine-grained control over what gets logged in each phase
- Provides flexibility to log only specific requests or responses based on various conditions
```elixir
plug Plug.LoggerJSON,
  should_log_request_fn: &MyApp.should_log_request/1,
  should_log_response_fn: &MyApp.should_log_response/1
```

#### 2. **Customizable Duration Units**
- Added support for three duration units: `:nanoseconds`, `:microseconds`, and `:milliseconds`
- Provides more precise timing measurements for performance monitoring
- Maintains backward compatibility with default millisecond logging
```elixir
plug Plug.LoggerJSON, duration_unit: :microseconds
```

#### 3. **Enhanced Exception Handling**
- Added `log_request/1` function for manual request logging in exception handlers
- Improves error tracking when normal logging flow is interrupted
- Ensures critical request information is captured even during failures

### 🔧 Improvements
- **Enhanced Type Specifications**: Improved type annotations for better development experience
- **Comprehensive Documentation**: Updated README with detailed examples and configuration options
- **Better Option Handling**: Refined internal option processing and validation
- **Development Environment Updates**: Updated dependencies and Elixir version configurations

### 📚 Documentation Updates
- Added comprehensive examples for all new features
- Documented separate request/response logging patterns and use cases
- Provided clear examples for duration unit configuration
- Enhanced inline documentation with better type information

### 🧪 Testing
- Added comprehensive test coverage for all new features
- Verified backward compatibility with existing configurations
- Tested edge cases and error scenarios
- Added tests for separate request/response logging
- Added tests for duration unit configurations
- Added tests for exception handling scenarios

### 💥 Breaking Changes
None. All changes are backward compatible with existing configurations.

### 🔄 Migration Notes
No migration required. All new features are opt-in with sensible defaults that maintain current behavior.

### Related Issues
- Addresses CX-40350: Enhanced logging capabilities for better observability

This PR significantly enhances the flexibility and power of `Plug.LoggerJSON` while maintaining full backward compatibility, making it more suitable for production environments with diverse logging requirements.

### CODE REVIEW

File: lib/plug/logger_json.ex [hunk 1]
```diff
@@ -55,44 +80,168 @@ defmodule Plug.LoggerJSON do
...
+      log(conn, level, start, opts)
+    end
+
+    # Register before_send callback for response phase
     Conn.register_before_send(conn, fn conn ->
-      :ok = log(conn, level, start, opts)
+      if should_log_response?(conn, opts) do
+        # Mark that we're in the before_send callback
+        conn = Conn.put_private(conn, :plug_logger_json_before_send, true)
+        log(conn, level, start, opts)
```
Comment (DFilipeS) [inline lib/plug/logger_json.ex]: The `plug_logger_json_before_send` is only needed on the `log/4` call? The `conn` that will be returned from this function won't have that field set.

Comment (DFilipeS) [reply lib/plug/logger_json.ex]: Should we keep the `:ok = ...`?

File: lib/plug/logger_json.ex [hunk 2]
```diff
@@ -149,14 +304,38 @@ defmodule Plug.LoggerJSON do
...
+        duration_microseconds
+
+      :milliseconds ->
+        # Convert microseconds to milliseconds and round to 3 decimal places
+        Float.round(duration_microseconds / 1000, 3)
+
+      _ ->
+        # Default to milliseconds for invalid values
+        Float.round(duration_microseconds / 1000, 3)
+    end
```
Comment (DFilipeS) [inline lib/plug/logger_json.ex]: Could this be done with `System.convert_time_unit`? https://hexdocs.pm/elixir/System.html#convert_time_unit/3

File: lib/plug/logger_json.ex (hunk 1 above)
Comment (eltonplima) [reply lib/plug/logger_json.ex]: The `plug_logger_json_before_send` is only needed to track the phase. [I updated this logic here to become clearer](a8f9e7b804e22a8501e6b025005ccf860cb924c0).

File: lib/plug/logger_json.ex (hunk 2 above)
Comment (eltonplima) [reply lib/plug/logger_json.ex]: [Done](e63a4cdb2311b9c1569e111078f0133a51d03e0e)

**Review by DFilipeS — APPROVED** [review]

**Review by mattheusmarzola — APPROVED** [review]
